dapos/dapos_handshake.go

```diff
@@ -185,7 +185,13 @@ func (this *DAPoSService) gossipWorker() {
 						this.gossipQueue.Push(gossip)
 
 						go func() {
-							time.Sleep(types.GossipQueueTimeout)
+							//adding timeout as a function of tx time.  If tx is in the future, add future delta to the default timeout
+							delta := gossip.Transaction.Time - utils.ToMilliSeconds(time.Now())
+							timeout := types.GossipQueueTimeout
```

Comment:
Rather than hard-code the GossipQueueTimeout, can we make it based on `(GossipTimeout` x # of delegates) + TxReceiveTimeout`? That way, we don't have to worry about the number of delegates we can deploy without breaking the logic.

Reply:
I have made that change